CHANGELOG.md

```diff
@@ -5,6 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
+- Add support for simulating events on Components [#211](https://github.com/preactjs/enzyme-adapter-preact-pure/pull/211)
```

Comment:
Since components are not DOM nodes, and this adapter dispatches real events, it might not be obvious what it means to simulate an event on a component. It seems to me there could be several options:

1. The implementation you have chosen here, where some DOM descendant is chosen and an event fired on that (if so, which one?).
2. Call the `onClick` prop on the component, and don't simulate a real event. I think this is what Enzyme does?

Approach (1) is potentially brittle as changing the internal DOM structure of a component might break tests in unexpected ways. For example if a `FancyButton` component renders a `<button>` with an `onClick` handler, then the `<button>` is wrapped in a `<div>` for layout/styling reasons, the `onClick` event would now be dispatched to the "wrong" element. The same event handler would get invoked, and the event would still bubble through DOM rendered by `FancyButton`'s parents, but the target of the event would be different.

Reply:
OK, I understand that following Enzyme's behavior is helpful for migrating a large existing test suite. Perhaps we could add some kind of suppressible warning in future to assist with migrations away from non-recommended coding patterns. Since the behavior isn't obvious, can you add a note in the `simulateEvent` method docs to explain why it has the current behavior.

Reply:
> I've updated the changelog to describe the behavior of the new feature flag. Let me know if that is what you had in mind.

This looks good, although I think we should spell out louder that this is a behavior change for host nodes when shallow rendering is enabled, since that is the change in behavior that may affect existing users.

Reply:
Add a paragraph describing the change to host nodes when this flag is turned on. I'm gonna complete the PR now so i can start building on top of this work, but if you have any further comments do add them and @-mention me! I'll happily address them in a follow up PR